src/build.sh

```diff
@@ -36,14 +38,15 @@ fi
 pushd tmx
 cmake -Bbuild -DCMAKE_PREFIX_PATH=\"/usr/local/share/tmx\;\/opt/carma/cmake\;\" -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_BUILD_TYPE="${BUILD_TYPE}" .
 pushd build
-make
+make -j${numCPU}
 make install
 popd
 popd
 
 pushd v2i-hub
 cmake -Bbuild  -DCMAKE_PREFIX_PATH=\"/usr/local/share/tmx\;\/opt/carma/cmake\;\" -DqserverPedestrian_DIR=/usr/local/share/qserverPedestrian/cmake -Dv2xhubWebAPI_DIR=/usr/local/share/v2xhubWebAPI/cmake/ -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_BUILD_TYPE="${BUILD_TYPE}" .
 pushd build
+make -j${numCPU}
```

Comment:
Why do we need this ${numCPU}? Whese is this env variable from?

Reply:
The env is for finding the number or cores available on the local machine. This allows for building in parallel. Since we're using Ubuntu as the docker image, this command to get the env variable will always work.

Reply:
Do you have a link explain how this command "make -j${numCPU}" allow building in parallel. Sorry, I have not seen this before and am curious how this works. Also if we are running this inside a base image ubuntu, how does that get the host machine hardware specs, like num of CPUs?

Reply:
Yep! Section 5.4 here: https://www.gnu.org/software/make/manual/make.html#Parallel

Yeah, it's still linked to the host machine, so it's able to find the number of cores available. Then this would run one job per core.

Reply:
If there is nothing looking like an integer after the ‘-j’ option, there is no limit on the number of job slots.

Reply:
![image](https://github.com/usdot-fhwa-OPS/V2X-Hub/assets/62157949/f0d02fa8-d025-47f2-88b6-2cb47c4ac583)

Reply:
Oh, numCPU set by `numproc`